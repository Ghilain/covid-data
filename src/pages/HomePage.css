.countrycontainer {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  margin-left: -39px;
}

.countrycontainer li:nth-child(4n),
.countrycontainer li:nth-child(4n + 1) {
  background-color: #172a2a;
}

.worldmap {
  width: 50%;
  margin-left: 25%;
  margin-top: 22px;
  height: inherit;
}

.imgcontainer {
  width: inherit;
  background-color: #638378;
  height: 300px;
}

.countcases {
  text-align: center;
}

.linkicon {
  font-size: 22px;
  color: aliceblue;
}

.countrycard {
  list-style: none;
  background-color: rgb(24 74 74);
}

.countryinfo {
  font-size: 10px;
  margin-left: 9px;
  margin-bottom: 9px;
}

.countryflag {
  width: 61%;
}

section {
  margin-bottom: 50px;
}

.flagcont {
  display: flex;
  justify-content: space-between;
  margin-left: 9px;
  margin-right: 9px;
  margin-top: 9px;
}

input {
  width: 96%;
  margin-left: 1%;
  font-size: 17px;
}
