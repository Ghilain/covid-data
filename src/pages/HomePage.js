import React, { useEffect, useState } from 'react';
import { Link, useNavigate, useLocation } from 'react-router-dom';
import { useSelector, useDispatch } from 'react-redux';
import { BiRightArrowCircle } from 'react-icons/bi';
import FetchData from '../redux/fetchapidata';
import { GetData } from '../redux/appreducer';
import './HomePage.css';

function HomePage() {
  const CountryData = useSelector((fetch) => fetch.moredetails);
  const dispatch = useDispatch();

  useEffect(() => {
    if (CountryData.length === 0) {
      FetchData()
        .then((response) => dispatch(GetData(response)));
    }
  }, []);

  let CountryHighCases = CountryData.filter((item) => item.country_cases > 10000000);
  const currentlocation = useLocation();
  const params = new URLSearchParams(currentlocation.search);
  const search = params.get('search') || '';
  CountryHighCases = CountryHighCases.filter((countries) => countries.country_name.toString().split('').join('')
    .toUpperCase()
    .toLowerCase()
    .replace(/\s/g, '')
    .includes(search));
  const navigation = useNavigate();
  const [searchData, setSearchData] = useState(search);

  const countryOnChange = (e) => {
    navigation(e.target.value ? `?search=${e.target.value}` : '');
    setSearchData(e.target.value);
  };

  return (
    <section>
      <h1>Countries with high cases</h1>
      <form>
        <input
          type="text"
          value={searchData}
          placeholder="Search for country..."
          onChange={countryOnChange}
        />
      </form>
      <ul className="countrycontainer">
        {
        CountryHighCases.map((countries) => (
          <li key={countries.country_name}>
            <Link key={countries.country_name} to={{ pathname: `/country/${countries.country_name}` }}>
              <BiRightArrowCircle />
            </Link>
            <div>
              <h3>
                Country:
                <br />
                {countries.country_name}
              </h3>
              <h3>
                Continent:
                <br />
                {countries.country_continent}
              </h3>
              <h3>
                Population:
                <br />
                {countries.population.toLocaleString()}
              </h3>
            </div>
            <div>
              <img src={countries.country_flag} alt="country flag" />
            </div>
          </li>
        ))
      }
      </ul>
    </section>
  );
}

export default HomePage;
