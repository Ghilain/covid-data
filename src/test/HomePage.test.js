import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/configureStore';
import HomePage from '../pages/HomePage';

describe('Test Pages', () => {
  test('test Homepage', () => {
    const tree = renderer
      .create(<Provider store={store}><BrowserRouter><HomePage /></BrowserRouter></Provider>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
