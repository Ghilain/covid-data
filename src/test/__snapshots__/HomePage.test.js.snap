// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Pages test Homepage 1`] = `
<section>
  <div
    className="imgcontainer"
  >
    <img
      alt="world map"
      className="worldmap"
      src="worldcovidmap.png"
    />
  </div>
  <h1
    className="countcases"
  >
    Countries with high cases
  </h1>
  <form>
    <input
      onChange={[Function]}
      placeholder="Search for country..."
      type="text"
      value=""
    />
  </form>
  <ul
    className="countrycontainer"
  />
</section>
`;
